src/components/PhotoSwipe/PhotoSwipe.tsx

```diff
@@ -527,6 +543,100 @@ function PhotoSwipe(props: Iprops) {
+                livePhotoVideo.style.opacity = 0;
+                livePhotoImage.style.opacity = 1;
+            };
+            setLivePhotoBtn(
+                <LivePhotoBtn
+                    onClick={() => {
+                        const video = document.getElementsByClassName(
+                            `live-photo-video-${item.id}`
+                        )[0];
+                        const image = document.getElementsByClassName(
+                            `live-photo-image-${item.id}`
+                        )[0];
+                        if (video && image) {
+                            playVideo(video, image);
+                        }
+                    }}
+                    onMouseEnter={() => {
+                        const video = document.getElementsByClassName(
+                            `live-photo-video-${item.id}`
+                        )[0];
+                        const image = document.getElementsByClassName(
+                            `live-photo-image-${item.id}`
+                        )[0];
+                        if (video && image) {
+                            playVideo(video, image);
+                        }
+                    }}
+                    onMouseLeave={() => {
+                        const video = document.getElementsByClassName(
+                            `live-photo-video-${item.id}`
+                        )[0];
+                        const image = document.getElementsByClassName(
+                            `live-photo-image-${item.id}`
+                        )[0];
+                        if (video && image) {
+                            pauseVideo(video, image);
+                        }
+                    }}>
+                    {livePhotoBtnHTML} LIVE
+                </LivePhotoBtn>
+            );
+
+            const downloadMessageDivs = document.getElementsByClassName(
+                `download-message`
+            ) as HTMLCollectionOf<HTMLDivElement>;
+            for (const downloadMessageDiv of downloadMessageDivs) {
+                if (downloadMessageDiv?.dataset?.id === item.id) {
+                    const downloadLivePhotoBtn =
+                        downloadMessageDiv.firstElementChild as HTMLButtonElement;
+
+                    const downloadLivePhoto = () => {
+                        downloadFileHelper(photoSwipe.currItem);
+                    };
+
+                    downloadLivePhotoBtn.addEventListener(
+                        'click',
+                        downloadLivePhoto
+                    );
+                    return () => {
+                        downloadLivePhotoBtn.removeEventListener(
+                            'click',
+                            downloadLivePhoto
+                        );
+                    };
+                }
+            }
+        }
+
+        return () => {
+            setLivePhotoBtn(null);
+        };
+    }, [photoSwipe?.currItem, isOpen, isSourceLoaded]);
```

Comment:
do we need isSourceLoaded here as the `photoSwipe?.curr.item` is anyways updated when the URL is changed. ?

Is there some case you came across when it doesn't happen ?

Reply:
Yeah, it doesn't seem to trigger the useEffect when the `item.html` or `item.url` changes, but it was good when the current item is changed though.

Reply:
I think this is fixed in the fix loading PR I have opened...will check after that... can ignore for now

Reply:
We still need that, if `isSourceLoaded` is removed the download button is not updated in case of no video playback.

Reply:
did you test it out after merging master? does it work without `isSourceLoaded` now?

Reply:
Yes, I did. The video and image element works without that but not the download button.

Reply:
cools... let's keep it then